crates/virtio-queue/src/mock.rs

```diff
@@ -368,17 +371,29 @@ impl<'a, M: GuestMemory> MockSplitQueue<'a, M> {
         q
     }
 
-    /// Writes a descriptor chain to the memory object of the queue and returns the associated
-    /// `DescriptorChain` object.
-    // `descs` represents a slice of `Descriptor` objects which are used to populate the chain.
+    /// Writes a single descriptor chain to the memory object of the queue, at the beginning of the
+    /// descriptor table, and returns the associated `DescriptorChain` object.
+    pub fn build_desc_chain(&self, descs: &[Descriptor]) -> DescriptorChain<&M> {
```

Comment:
Why do we still need this function? It is not clear to me how you have to use build_desc_chain and add_desc_chain. From the name it sounds like you can use it as:
- first create a descriptor chain using `build_desc_chain`
- then you can add more descriptors using `add_desc_chain`
Is this assumption correct?

Reply:
Not necessarily, the `build` one is also returning that chain by iterating the queue, the second one can be used to simply populate the queue with a new chain. I kept the old one because it helps in tests to also have that chain returned. The naming is pretty bad, but since I didn't necessarily have a much better idea of a name for this one, I chose to at least keep it the same as before so that there are no other name changes involved in the PR.

Reply:
You can use `add_desc_chain` without any `build_desc_chain` call. Basically what `build_desc_chain` does differently is that is also starts to iterate over the queue.

Reply:
I find this to be a bit strange, but let's just merge it an fix it when we're extending the mocking framework.